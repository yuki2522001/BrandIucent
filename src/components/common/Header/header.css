.header {
  position: relative;
  z-index: 20;
  background: var(--bg-first);
}

.header__info {
  padding: 207px 0 257px 162px;
}

.header__title {
  color: var(--dark);
  font-weight: 600;
  font-size: 36px;
  line-height: 54px;
  width: 515px;
  margin: 16px 0 47px;
}

.header__images--right {
  position: absolute;
  z-index: 10;
  top: 0;
  right: 0;
}

.header__images--basket {
  position: absolute;
  z-index: 15;
  top: 18px;
  right: 264px;
}

.header__images--top-view {
  position: absolute;
  z-index: 10;
  top: 105px;
  right: 0;
}