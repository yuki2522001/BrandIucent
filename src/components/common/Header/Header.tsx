import React from "react";
import Button from "../Button/Button";
import Text from "../Text/Text";
import NavigationBar from "../NavigationBar/NavigationBar";
import backgroundGreen from "@assets/images/backgrounds/green.png";
import backgroundBasket from "@assets/images/backgrounds/basket-full.png";
import backgroundTopView from "@assets/images/backgrounds/top-view.png";
import "./header.css";

const Header: React.FC = () => {
  return (
    <header data-testid="header" className="header">
      <NavigationBar mode="light" />
      <div className="header__info">
        <Text text="Lorem ipsum dolor sit" decoration="blur" />
        <h1 className="header__title">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </h1>
        <Button text="Sign Up" type="large" />
      </div>
      <figure>
        <img
          className="header__images--right"
          src={backgroundGreen}
          alt="This is background green color"
        />
        <img
          className="header__images--basket"
          src={backgroundBasket}
          alt="This is background basket-full"
        />
        <img
          className="header__images--top-view"
          src={backgroundTopView}
          alt="This is background top-view"
        />
      </figure>
    </header>
  );
};

export default Header;
