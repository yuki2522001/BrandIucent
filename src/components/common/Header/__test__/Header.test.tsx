import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "@testing-library/jest-dom";
import Header from "../Header";

describe("Header component", () => {
  test("should render header component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );
    expect(getByTestId("header")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
