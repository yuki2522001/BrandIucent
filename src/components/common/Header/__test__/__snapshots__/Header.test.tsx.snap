// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
    data-testid="header"
  >
    <nav
      class="nav"
      data-testid="navigation-bar"
    >
      <a
        data-testid="logo"
        href="/"
      >
        <img
          alt="This is logo"
          class=""
        />
      </a>
      <div
        class="nav__about"
      >
        <div
          class="menu"
          data-testid="menu"
        >
          <ul
            class="menu__list"
          >
            <li
              class="menu__item"
              data-testid="menu-item"
            >
              <a
                href="/products"
              >
                Shop
              </a>
            </li>
            <li
              class="menu__item"
            >
              <a
                href="/"
              >
                Orders
              </a>
            </li>
            <li
              class="menu__item"
            >
              <a
                href="/"
              >
                Cart
              </a>
            </li>
            <li
              class="menu__item"
            >
              <a
                href="/"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
        <button
          class="btn btn--light"
        >
          Sign Up
        </button>
        <button
          class="btn btn__outline--light"
        >
          Login
        </button>
      </div>
    </nav>
    <div
      class="header__info"
    >
      <p
        class="text--blur"
        data-testid="text"
      >
        Lorem ipsum dolor sit
      </p>
      <h1
        class="header__title"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </h1>
      <button
        class="btn btn--large"
      >
        Sign Up
      </button>
    </div>
    <figure>
      <img
        alt="This is background green color"
        class="header__images--right"
      />
      <img
        alt="This is background basket-full"
        class="header__images--basket"
      />
      <img
        alt="This is background top-view"
        class="header__images--top-view"
      />
    </figure>
  </header>
</DocumentFragment>
`;
